core/templates/dev/head/domain/utilities/UrlInterpolationServiceSpec.js

```diff
@@ -299,6 +299,10 @@ describe('URL Interpolation Service', function() {
       hashes['/path_test/hash_test.html'] + '.html');
   });
 
+  it('should interpolate URLs not requiring parameters', function() {
+    expect(uis.getStoryUrl('/storyId', {})).toBe('/story/storyId');
```

Comment:
You should have more tests than this. What about unicode chars? Problematic cases? Empty args? Etc.

Don't think about the specific implementation that's currently used (it may change), think about the interface and test it to make sure it's robust.

Reply:
Added more tests